pkg/app/app.go

```diff
@@ -69,9 +70,14 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 		return err
 	}
 
+	// ensure the artifacts dir
+	if err := os.MkdirAll(artifacts.BaseDir(), os.ModePerm); err != nil {
+		return err
+	}
+
 	// setup junit writer
 	junitRunner, err := os.Create(
-		filepath.Join(opts.RunDir(), "junit_runner.xml"),
```

Comment:
Currently, `RunDir` is a sub directory to ARTIFACTS directory. Hence the above location of `junit_runner.xml` in RunDir would definitely upload the file to ARTIFACTS.

But as this change is to decouple RunDir from Artifacts, and file `junit_runner.xml` is an artifact file rather than one specific to each run of kubetest2, hence I have made the change of location from `opts.RunDir()` to `artifacts.BaseDir()`